src/Components/ArticleCard.jsx

```diff
@@ -9,13 +9,32 @@ const ArticleCard = () => {
     const [article, setArticle] = useState({})
     const {article_id} = useParams('article_id')
     const [isLoading, setIsLoading] = useState(true)
+    const [votes, setVotes] = useState('')
+    const [err, setErr] = useState(null);
 
     useEffect(() => {
         getArticleByID(article_id).then((article) => {
             setArticle(article)
+            setVotes(article.votes)
         })
         setIsLoading(false)
     }, [])
+    const handleUpVote = (event) => {
+        setVotes((currentVote) => currentVote +1);
+        setErr(null)
+        patchArticle(1, article_id).catch((err) => {
+            setVotes((currentVote) => currentVote -1)
+            setErr(err)
+        })
+    }
+    function handleDownVote(event){
+        setVotes((currentVote) => currentVote -1);
+        setErr(null)
+        patchArticle(-1, article_id).catch((err) => {
+            setVotes((currentVote) => currentVote +1)
+            setErr(err)
+        })
```

Comment:
Absolutely love this logic! Well done on also considering what should happen if the vote didn't work, e,g if the internet went off!